### PR TITLE
feat(bivariate-color-manager) 11506 - fix maps count calculation

### PR DESCRIPTION
**Pull request by propakov**
https://kontur.fibery.io/Tasks/Task/The-number-of-Maps-should-be-considered-as-a-multiplication-of-maps-on-the-two-axes-11506
before
Screenshot 2022-08-11 at 16 40 46
after
Screenshot 2022-08-11 at 16 41 31

### CODE REVIEW

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts [hunk 1]
```diff
@@ -99,39 +103,46 @@ export const bivariateColorManagerResourceAtom = createResourceAtom(
...
+        if (!acc[numerator]) {
+          acc[numerator] = {
+            denominator,
+            quality: axis.quality,
+            layersCount: 1,
+          };
+        } else {
+          acc[numerator].layersCount++;
+
+          const prevAxisQualityByNumerator = acc[numerator].quality;
```
Comment (Akiyamka) [inline src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts:119]: why it "prev"?

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts [hunk 2]
```diff
@@ -99,39 +103,46 @@ export const bivariateColorManagerResourceAtom = createResourceAtom(
...
+          if (prevAxisQualityByNumerator < axis.quality) {
+            acc[numerator] = {
+              ...acc[numerator],
+              denominator,
+              quality: axis.quality,
+            };
           }
```
Comment (Akiyamka) [inline src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: So, we looking for most quality axis here?

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts [hunk 3]
```diff
@@ -241,15 +246,24 @@ export const bivariateColorManagerResourceAtom = createResourceAtom(
...
   Object.values(bivariateColorManagerData).forEach((row) => {
-    [...Object.values(row.horizontal), ...Object.values(row.vertical)].forEach(
-      (layer) => {
-        layer.correlationLevel = numeratorCorellationMap[layer.name] || 0;
-      },
-    );
+    let horizontalLayersSum = 0;
+    let verticalLayersSum = 0;
+
+    [...Object.values(row.horizontal)].forEach((layer) => {
```
Comment (Akiyamka) [inline src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: Why not just `Object.values(row.horizontal).forEach` ?

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts (hunk 2 above)
Comment (propakov) [reply src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: most quality denominator
it looks like it's better to add comment, isn't it?  🤔

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts (hunk 3 above)
Comment (propakov) [reply src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: Nice catch! It's just stayed after refactoring 🤦

File: src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts (hunk 2 above)
Comment (propakov) [reply src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: I've decided to change naming to be clear

acc[numerator] = {
            mostQualityDenominator: denominator,
            quality: axis.quality,
            layersCount: 1,
          };
          
i think it's clear that for numerator you get mostQualityDenominator, quality, layersCount
does it look ok? @Akiyamka

Comment (Akiyamka) [reply src/features/bivariate_color_manager/atoms/bivariateColorManagerResource.ts]: yep, looks better

**Review by Akiyamka — APPROVED** [review]

**Pull request merged**